extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/ConfigurationService.java

```diff
@@ -0,0 +1,314 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.auth.radius;
+
+import com.google.inject.Inject;
+import java.util.Collections;
+import java.util.List;
+import org.apache.guacamole.GuacamoleException;
+import org.apache.guacamole.environment.Environment;
+
+/**
+ * Service for retrieving configuration information regarding the RADIUS server.
+ */
+public class ConfigurationService {
+
+    /**
+     * The Guacamole server environment.
+     */
+    @Inject
+    private Environment environment;
+
+    /**
+     * Returns the hostname of the RADIUS server as configured with
+     * guacamole.properties. By default, this will be "localhost".
+     *
+     * @return
+     *     The hostname of the RADIUS server, as configured with
+     *     guacamole.properties.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties cannot be parsed.
+     */
+    public String getRadiusServer() throws GuacamoleException {
+        return environment.getProperty(
+            RadiusGuacamoleProperties.RADIUS_SERVER,
+            "localhost"
+        );
+    }
+
+    /**
+     * Returns the authentication port of the RADIUS server configured with
+     * guacamole.properties.
+     *
+     * @return
+     *     The authentication port of the RADIUS server, as configured with
+     *     guacamole.properties.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties cannot be parsed.
+     */
+    public int getRadiusAuthPort() throws GuacamoleException {
+        return environment.getProperty(
+            RadiusGuacamoleProperties.RADIUS_AUTH_PORT,
+            1812
+        );
+    }
+
+    /**
+     * Returns the accounting port of the RADIUS server configured with
+     * guacamole.properties. 
+     *
+     * @return
+     *     The accouting port of the RADIUS server, as configured with
+     *     guacamole.properties.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties cannot be parsed.
+     */
+    public int getRadiusAcctPort() throws GuacamoleException {
+        return environment.getProperty(
+            RadiusGuacamoleProperties.RADIUS_ACCT_PORT,
+            1813
+        );
+    }
+
+    /**
+     * Returns the shared secret of the RADIUS server configured with
+     * guacamole.properties. 
+     *
+     * @return
+     *     The shared secret of the RADIUS server, as configured with
+     *     guacamole.properties.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties cannot be parsed.
+     */
+    public String getRadiusSharedSecret() throws GuacamoleException {
+        return environment.getProperty(
+            RadiusGuacamoleProperties.RADIUS_SHARED_SECRET,
+            null
```

Comment:
There's no need to explicitly specify `null` as the default if there is no default. Just using:

    environment.getProperty(RadiusGuacamoleProperties.RADIUS_SHARED_SECRET)

has that behavior. It will return `null` if the property is not specified.